QuizApp/App/Screens/Quiz/QuizListViewModel.swift

```diff
@@ -17,32 +18,43 @@ final class QuizListViewModel {
 
 extension QuizListViewModel {
 
+    @MainActor
     func fetchQuiz(for category: CategoryModel) {
         Task {
-            let quizes = try await quizUseCase.fetchQuizes(for: category)
+            do {
+                let quizes = try await quizUseCase.fetchQuizes(for: category)
 
-            DispatchQueue.main.async { [weak self] in
-                guard let self = self else { return }
+                DispatchQueue.main.async { [weak self] in
+                    guard let self = self else { return }
 
-                self.quizes = quizes
-                    .map { QuizViewModel(from: $0) }
+                    self.quizes = quizes
+                        .map { QuizViewModel(from: $0) }
+                }
+            } catch {
+                hideEmptyStateView = false
             }
         }
     }
 
-    func fetchAllQuizes() {
-        Task {
-            let quizes = try await quizUseCase.fetchQuizes()
+    func fetchAllQuizes() throws {
+            Task {
+                do {
+                    let quizes = try await quizUseCase.fetchQuizes()
+
+                    DispatchQueue.main.async { [weak self] in
+                        guard let self = self else { return }
 
-            DispatchQueue.main.async { [weak self] in
-                guard let self = self else { return }
+                        self.quizes = quizes
+                            .map { QuizViewModel(from: $0) }
+                    }
+                } catch {
+                    hideEmptyStateView = false
+                }
 
-                self.quizes = quizes
-                    .map { QuizViewModel(from: $0) }
             }
-        }
     }
 
+    @MainActor
```

Comment:
Why is this @MainActor?

Reply:
As I've marked `fetchQuiz(for category: CategoryModel)` with` @MainActor` and that method calls `fetchInitialQuiz()`, compiler is not happy with an error `Call to main actor-isolated instance method 'fetchQuiz(for:)' in a synchronous nonisolated context`. 

I will refactor those methods in the sections PR, so will look into this more.